lib/lz4.c

```diff
@@ -1003,7 +1019,9 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
         *inputConsumed = (int) (((const char*)ip)-source);
     }
     DEBUGLOG(5, "LZ4_compress_generic: compressed %i bytes into %i bytes", inputSize, (int)(((char*)op) - dest));
-    return (int)(((char*)op) - dest);
+    result = (int)(((char*)op) - dest);
+    if (result <= 0) cctx->needsFullReset = 1;
```

Comment:
Is this actually needed? I believe at this point we are guaranteed success.

Reply:
Right, better to `assert(result>0)` instead.

Reply:
Done, thanks!